### PR TITLE
refactor/action: rewrite correct_lines purely based on spans

### LINKED ISSUE #116: Newlines are not preserved
**Describe the bug**

When the fixes are applied, all newlines are convert to `\n`.

**To Reproduce**

Steps to reproduce the behaviour:

1. A file containing `\n\r` or `\r`
2. Run `cargo spellcheck fix`
3. Check newlines from a capable editor

**Expected behavior**

Preserve the newlines that were there initially

**Please complete the following information:**
 - System:  Linux
 - Obtained:  git
 - Version:  git

### PR DESCRIPTION
**Pull request by drahnr**
## What does this PR accomplish?

Replaces the correction algorithm.

 * 🩹 Bug Fix
 * 🦚 Feature

Closes #116  .

## Changes proposed by this PR:

Previously a line based algortihm was employed which caused some issues in #95 and #116 .

The new implementation works purely on `Span`s wrapped in `Patch`es, which now also handles multiline replacements properly and forces a clear boundary between `Patch` and `Bandaid`.

## Notes to reviewer:

It's now entirely the concern of the checker to provide sane spans and proper wrapping with comment markers (i.e. `///` ) from within the `Checker`s - this is mostly necessary for #95 .

## 📜 Checklist

 * [x] Works on the `./demo` sub directory
 * [x] Test coverage is excellent and passes
 * [x] Documentation is thorough

### CODE REVIEW

**Comment by drahnr** [comment]
Will merge this if nobody reviews :jack_o_lantern:

File: src/action/mod.rs [hunk 1]
```diff
@@ -35,90 +35,176 @@ impl Finish {
...
+///
+/// This function is not concerend with _any_ semantics or comments or
+/// whatsoever at all.
+fn correct_lines<'s, II, I>(patches: II, source_buffer: String, mut sink: impl Write) -> Result<()>
+where
+    II: IntoIterator<IntoIter = I, Item = Patch>,
+    I: Iterator<Item = Patch>,
+{
+    let patches = patches.into_iter();
+    let mut patches = patches.peekable();
```
Comment (KuabeM) [inline src/action/mod.rs:92]: Nitpick: what's wrong with putting it together like this?

```suggestion
    let mut patches = patches.into_iter().peekable();
```

File: src/span.rs [hunk 2]
```diff
@@ -173,6 +173,30 @@ impl TryFrom<(usize, Range)> for Span {
     }
 }
 
+impl TryFrom<(usize, std::ops::RangeInclusive<usize>)> for Span {
```
Comment (KuabeM) [inline src/span.rs:176]: :+1:

**Review by KuabeM — DISMISSED** [review]
Looking good to me! Is a huge improvement for `reflow`..

**Pull request merged**